### PR TITLE
Publish Windows artifacts with manifest publishing

### PR DESCRIPTION
**Pull request by fheinecke**
This PR adds release server manifest publishing for our Windows artifacts. It needs to be merged at the same time as https://github.com/gravitational/teleport.e/pull/4412 to prevent breakage, and will also require an `e` ref update.

Note: There is a Windows-specific bug in `relcli` where it will include the full file path in the manifest, i.e. `C:\artifacts\teleport-windows-auth-setup-v17.0.0-dev.fred-win-manifests-6-amd64.exe`. I'll file a PR in the `cloud` repo to fix this, and that PR will need to land/be deployed prior to merge of this PR.

Passing build: https://github.com/gravitational/teleport.e/actions/runs/9542150214/job/26296480586

### CODE REVIEW

File: build.assets/windows/build.ps1 [hunk 1]
```diff
@@ -195,45 +174,53 @@ function Get-Relcli {
...
-        $env:DRONE_TAG = "$ArtifactVersion"
+    $SearchPath = Join-Path -Path $ArtifactDirectory -ChildPath *
+    Get-ChildItem -Path $SearchPath -Include "*.exe","*.zip" | ForEach-Object {
+        switch -Wildcard ($_.Name) {
+            "Teleport Connect Setup*.exe" {
+                $description = "Teleport Connect"
+                Break
+            }
+            "teleport-windows-auth-setup*.exe" {
+                $description = "Teleport Connect"
```
Comment (marcoandredinis) [inline build.assets/windows/build.ps1:206]: Should this have a different description?
Same for the other archive in the next switch case

**Review by marcoandredinis — COMMENTED** [review]
Why did we remove the sha256 hash for the files?

**Comment by fheinecke** [comment]
> Why did we remove the sha256 hash for the files?

The checksum is now stored inside the manifest, which is what's eventually sent to the release server

**Review by marcoandredinis — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Comment by fheinecke** [comment]
https://github.com/gravitational/github-terraform/pull/994 needs to land prior to merge, or download links will be broken upon the next release.

**Comment by fheinecke** [comment]
Passing workflow run: https://github.com/gravitational/teleport.e/actions/runs/9657728876/job/26637529033

**Pull request merged**